keps/sig-api-machinery/2144-clientgo-apply/README.md

```diff
@@ -159,69 +159,48 @@ take an aligned approach to adding apply to clients in a typesafe way.
 The client-go typed clients will be extended to include Apply functions, e.g.:
 
 ```go
-func (c *deployments) Apply(ctx Context, fieldManager string, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
-func (c *deployments) ApplyStatus(ctx Context, fieldManager string, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
-func (c *deployments) ApplyScale(ctx Context, fieldManager string, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
+func (c *deployments) Apply(ctx Context, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
+func (c *deployments) ApplyStatus(ctx Context, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
+func (c *deployments) ApplyScale(ctx Context, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
 ```
 
 `ApplyOptions` will be added to metav1 even though `PatchOptions` will continue
-to be used over the wire. This will make it obvious in the Apply function
-signature that `fieldManager` is required.
+to be used over the wire:
 
 ```go
 type ApplyOptions struct {
   DryRun []string `json:"dryRun,omitempty" protobuf:"bytes,1,rep,name=dryRun"`
-  Force *bool `json:"force,omitempty" protobuf:"varint,2,opt,name=force"`
-}
 
-func (ApplyOptions) ToPatchOptions(fieldManager string) PatchOptions
+  // <apply specific Force godoc goes here>
+  // Note that this is different than PatchOptions, which has Force as an optional field (bool pointer).
+  Force bool `json:"force,omitempty" protobuf:"varint,2,opt,name=force"`
+
+  // <apply specific FieldManager godoc goes here>
+  FieldManager string `json:"fieldManager,omitempty" protobuf:"bytes,3,name=fieldManager"`
+}
 ```
 
-<<[UNRESOLVED @jpbetz ]>>
+`ApplyOptions` is introduced to allow us to:
```

Comment:
Drawback is that you have to update the fields every time we update PatchOptions. That doesn't actually happen that frequently, so probably not a problem.